### PR TITLE
Implement Linux benchmark tool

### PR DESCRIPTION
**Pull request by pzread**
Implement Linux benchmark tool, similar to `run_benchmarks_on_android.py`.

### CODE REVIEW

File: build_tools/benchmarks/run_benchmarks_on_linux.py [hunk 1]
```diff
@@ -6,16 +6,140 @@
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 """Runs all matched benchmark suites on a Linux device."""
 
+import subprocess
+import atexit
+import os
+import re
+import shutil
+import sys
+import tarfile
```
Comment (antiagainst) [inline build_tools/benchmarks/run_benchmarks_on_linux.py:15]: Empty line after this.

File: build_tools/benchmarks/run_benchmarks_on_linux.py [hunk 2]
```diff
@@ -6,16 +6,140 @@
...
+from typing import Optional, Sequence
```
Comment (antiagainst) [inline build_tools/benchmarks/run_benchmarks_on_linux.py:16]: Empty line after this.

File: build_tools/benchmarks/run_benchmarks_on_linux.py [hunk 3]
```diff
@@ -6,16 +6,140 @@
...
+    tool_path = os.path.join(capture_config.traced_benchmark_tool_dir,
+                             tool_name)
+    cmd = ["taskset", taskset, tool_path, f"--flagfile={MODEL_FLAGFILE_NAME}"]
+    process = subprocess.Popen(cmd,
+                               env={"TRACY_NO_EXIT": "1"},
+                               cwd=case_dir,
+                               stdout=subprocess.PIPE,
+                               universal_newlines=True)
+
+    while True:
```
Comment (antiagainst) [inline build_tools/benchmarks/run_benchmarks_on_linux.py:82]: We have similar things in `run_benchmarks_on_android.py`. Can we unify them?

**Review by antiagainst — CHANGES_REQUESTED** [review]

File: build_tools/benchmarks/run_benchmarks_on_linux.py (hunk 2 above)
Comment (pzread) [reply build_tools/benchmarks/run_benchmarks_on_linux.py]: Added the empty line after `import tarfile`. Do we need to have a newline after `from typing import Optional, Sequence`?

File: build_tools/benchmarks/common/benchmark_definition.py [hunk 4]
```diff
@@ -99,6 +103,58 @@ def execute_cmd_and_get_output(args: Sequence[str],
...
+    ])
+  else:
+    cmd.extend([
+        f"--benchmark_repetitions={repetitions}",
+    ])
+
+  return cmd
+
+
+def wait_for_iree_benchmark_moudle_start(process: subprocess.Popen,
```
Comment (antiagainst) [inline build_tools/benchmarks/common/benchmark_definition.py:142]: Typo: module

File: build_tools/benchmarks/run_benchmarks_on_linux.py (hunk 2 above)
Comment (antiagainst) [reply build_tools/benchmarks/run_benchmarks_on_linux.py]: It's clearer that way given we are importing stuff from our own repo. :) But not a big deal. Your choice.

**Review by antiagainst — APPROVED** [review]

File: build_tools/benchmarks/run_benchmarks_on_linux.py (hunk 2 above)
Comment (pzread) [reply build_tools/benchmarks/run_benchmarks_on_linux.py]: I see. Done : )

**Pull request merged**